data_model/src/role.rs

```diff
@@ -19,7 +18,7 @@ use crate::{
 };
 
 /// Collection of [`RoleId`](Id)s
-pub type RoleIds = btree_set::BTreeSet<<Role as Identifiable>::Id>;
+pub type RoleIds = PHashTrieSet<<Role as Identifiable>::Id>;
```

Comment:
I think we should not use *Hash-based* structures in user-facing places, because hash function can be changed in different versions and either be incompatible between different platforms

Reply:
But when we store and transfer such structure we never send _hash_ (in scale we just serialize it as `k1,v1,k2,v2,...`), so this shouldn't be a problem, no?

Reply:
The problem is that users may rely on sorted order. I thought there was a reason to use `BTreeSet` here...

Reply:
Hmm, idk... we have sorting in `torii` if user wants sorted collection, no?

Reply:
AFAIK this maps are never part of transaction or block. 
But this change wouldn't be in the PR. So it doesn't really meter.

Reply:
Yes, I'm saying this change is ok. You can use hash map here